### ISSUE TITLE
Support for Better Than Adventure

### ISSUE THREAD

**Issue by Atemo-C** [Daxanius/NoMiningCooldown#3]
Greetings. Firstly, I want to thank you for this continuation of the mod, it is an absolute must-have for me and many players in the community.

---

### Better Than Adventure support
I was wondering: Is there was any way to make this mod work for the [Better Than Adventure](https://www.minecraftforum.net/forums/mapping-and-modding-java-edition/minecraft-mods/3106066-better-than-adventure-for-beta-1-7-3-timely) “conversion” of Minecraft?

### Fabric/Babric
[Babric](https://babric.github.io/), a fork of Fabric made to run mods on Minecraft Beta 1.7.3 *(on which Better Than Adventure is based on)*, can be used to run mods on this version. See [this page](https://bta.miraheze.org/wiki/Modding) for more details.

I personally use the [Prism Launcher](https://prismlauncher.org/), which supports it out of the box.

### A true life-saviour
No Mining Cooldown has been a life saviour for me in modern Minecraft, and in this version, it would be even more of one, as there are no such things as enchantments. Rather, each material acts differently in tools, (for example, Diamond tools mine the fastest, but Steel tools have the most durability), but the arbitrary mining cooldown makes it so that there is almost no instant-mining, unless one precisely times each hold and delay in milliseconds between mouse clicks.

### Minimal porting
Porting even just the very basics of this mod (just removing the cooldown without any option) would honestly be great, but I understand if this would not be possible or simply not worth the effort for such a minuscule niche of Minecraft. Thank you!

**Comment by Daxanius** [comment]
BTA looks very interesting! I might play it myself with some friends :)

In any case, I'll look into this. It doesn't seem like it'd be too much effort.

**Comment by Daxanius** [comment]
It seems like the [Babric mappings](https://github.com/babric/barn) haven't been updated in at least a year.

There's a lot of obfuscated code to work with which makes this task less than ideal.

**Comment by Atemo-C** [comment]
Yeah, this is not ideal, though a lot of mods still are being released almost on the daily. This is not a priority by any mean, so feel free to put in any or no time into this. Also, I am glad that BTA looks interesting to you! The next major release is also very exciting.

**Comment by Flamarine** [comment]
> It seems like the [Babric mappings](https://github.com/babric/barn) haven't been updated in at least a year.
> 
> There's a lot of obfuscated code to work with which makes this task less than ideal.

~~well this a bit late~~

bta does not obfuscate its code back to match vanilla so I don't think that's a problem?

**Comment by Daxanius** [comment]
Can you please elaborate what you mean with "does not obfuscate its code back to match vanilla"? I don't think I understand what you're trying to say. 

As to my knowledge, BTA is a mod, not a mod-loader. I do not see the relation between BTA and this?

**Comment by Flamarine** [comment]
> Can you please elaborate what you mean with "does not obfuscate its code back to match vanilla"? I don't think I understand what you're trying to say. 
> 
> As to my knowledge, BTA is a mod, not a mod-loader. I do not see the relation between BTA and this?

what I mean is normally a jar mod would obfuscate its code back to vanilla state (for copyright issues, better patching, etc), but in BTA we don't do that. all classes, methods and fields are named.

and actually, there does exist [an example mod](https://github.com/Turnip-Labs/bta-example-mod) for bta so that'd be a good kickstart, you can see it makes no use of mappings as we don't need them

**Comment by Daxanius** [comment]
I see, I wasn't aware of that. I will give it another shot as soon as I have some time for it!
